crates/proof-of-sql/src/sql/postprocessing/group_by_postprocessing.rs

```diff
@@ -169,6 +177,153 @@ impl GroupByPostprocessing {
     }
 }
 
+impl<S: Scalar> PostprocessingStep<S> for GroupByPostprocessing {
+    /// Apply the group by transformation to the given `OwnedTable`.
+    fn apply(&self, owned_table: OwnedTable<S>) -> PostprocessingResult<OwnedTable<S>> {
+        // First evaluate all the aggregated columns
+        let alloc = Bump::new();
+        let evaluated_columns: HashMap<AggregationOperator, Vec<(Identifier, OwnedColumn<S>)>> =
+            self.aggregation_expr_map
+                .iter()
+                .map(
+                    |((agg_op, expr), id)| -> PostprocessingResult<(
+                        AggregationOperator,
+                        (Identifier, OwnedColumn<S>),
+                    )> {
+                        let evaluated_owned_column = owned_table.evaluate(expr)?;
+                        Ok((*agg_op, (*id, evaluated_owned_column)))
+                    },
+                )
+                .collect::<PostprocessingResult<Vec<_>>>()?
+                .into_iter()
+                .into_group_map();
+        // Next actually do the GROUP BY
+        let group_by_ins = self
+            .group_by_identifiers
+            .clone()
+            .into_iter()
+            .map(|id| {
+                let column = owned_table
+                    .inner_table()
+                    .get(&id)
+                    .ok_or(PostprocessingError::ColumnNotFound(id.to_string()))?;
+                Ok(Column::<S>::from_owned_column(column, &alloc))
+            })
+            .collect::<PostprocessingResult<Vec<_>>>()?;
+        // TODO: Allow a filter
+        let selection_in = vec![true; owned_table.num_rows()];
+        let (sum_ids, sum_ins): (Vec<_>, Vec<_>) = evaluated_columns
+            .get(&AggregationOperator::Sum)
+            .map(|tuple| {
+                tuple
+                    .iter()
+                    .map(|(id, c)| (*id, Column::<S>::from_owned_column(c, &alloc)))
+                    .unzip()
+            })
+            .unwrap_or((vec![], vec![]));
+        let (max_ids, max_ins): (Vec<_>, Vec<_>) = evaluated_columns
+            .get(&AggregationOperator::Max)
+            .map(|tuple| {
+                tuple
+                    .iter()
+                    .map(|(id, c)| (*id, Column::<S>::from_owned_column(c, &alloc)))
+                    .unzip()
+            })
+            .unwrap_or((vec![], vec![]));
+        let (min_ids, min_ins): (Vec<_>, Vec<_>) = evaluated_columns
+            .get(&AggregationOperator::Min)
+            .map(|tuple| {
+                tuple
+                    .iter()
+                    .map(|(id, c)| (*id, Column::<S>::from_owned_column(c, &alloc)))
+                    .unzip()
+            })
+            .unwrap_or((vec![], vec![]));
+        let count_ids: Vec<_> = evaluated_columns
+            .get(&AggregationOperator::Count)
+            .unwrap_or(&vec![])
+            .iter()
+            .map(|(id, _)| *id)
+            .collect::<Vec<_>>();
```

Comment:
We shouldn't need to collect this.
```suggestion
            .map(|(id, _)| *id);
```